### PR TITLE
task(iOS): update dropbox authentication to PKCE flow

### PR DESCRIPTION
**Pull request by alexbumbu**

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -297,7 +297,7 @@ PODS:
     - react-native-video/Video (= 5.1.1)
   - react-native-video/Video (5.1.1):
     - React-Core
-  - react-native-webrtc (1.92.1):
```
Comment (saghul) [inline ios/Podfile.lock:300]: Looks like you need to rebase.

File: ios/sdk/src/dropbox/Dropbox.m [hunk 2]
```diff
@@ -128,25 +131,31 @@ + (BOOL)application:(UIApplication *)app
...
-    if (authResult) {
-        if ([authResult isSuccess]) {
-            currentResolve(authResult.accessToken.accessToken);
-        } else {
-            NSString *msg;
-            if ([authResult isError]) {
-                msg = [NSString stringWithFormat:@"%@, error type: %zd",[authResult errorDescription], [authResult errorType]];
+    
+    BOOL canHandle = [DBClientsManager handleRedirectURL:url completion:^(DBOAuthResult *authResult) {
+        if (authResult) {
```
Comment (saghul) [inline ios/sdk/src/dropbox/Dropbox.m:136]: Is it possible to get nil here? What does it mean?

File: ios/sdk/src/dropbox/Dropbox.m [hunk 3]
```diff
@@ -128,25 +131,31 @@ + (BOOL)application:(UIApplication *)app
...
+                                           @"rToken": authResult.accessToken.refreshToken,
+                                           @"expireDate": @(msTimestamp)
+                };
+                currentResolve(authInfo);
             } else {
-                msg = @"OAuth canceled!";
+                NSString *msg;
+                if ([authResult isError]) {
+                    msg = [NSString stringWithFormat:@"%@, error type: %zd", [authResult errorDescription], [authResult errorType]];
+                } else {
```
Comment (saghul) [inline ios/sdk/src/dropbox/Dropbox.m:148]: How is it possible to get here? Both `authResult isSuccess` and `uthResult isError` would need to return false, which feels weird.

**Review by saghul — COMMENTED** [review]
A couple of comments, PTAL and rebase.

File: ios/sdk/src/dropbox/Dropbox.m (hunk 2 above)
Comment (alexbumbu) [reply ios/sdk/src/dropbox/Dropbox.m:136]: I don't know. That's how the previous implementation was done and that's how it's done in the dropbox documentation example.

Comment (saghul) [reply ios/sdk/src/dropbox/Dropbox.m:136]: Ack.

File: ios/sdk/src/dropbox/Dropbox.m (hunk 3 above)
Comment (alexbumbu) [reply ios/sdk/src/dropbox/Dropbox.m:148]: Again, this is how the previous implementation was done. This sample code is from the dropbox documentation. Should answer your question:

```
if (authResult != nil) {
      if ([authResult isSuccess]) {
        NSLog(@"\n\nSuccess! User is logged into Dropbox.\n\n");
      } else if ([authResult isCancel]) {
        NSLog(@"\n\nAuthorization flow was manually canceled by user!\n\n");
      } else if ([authResult isError]) {
        NSLog(@"\n\nError: %@\n\n", authResult);
      }
```

Comment (saghul) [reply ios/sdk/src/dropbox/Dropbox.m:148]: Ah, there is an isCancel, ok!

File: ios/Podfile.lock (hunk 1 above)
Comment (alexbumbu) [reply ios/Podfile.lock:300]: weird, it's up to date

Comment (alexbumbu) [reply ios/Podfile.lock]: fixed

**Review by saghul — APPROVED** [review]

**Pull request merged**